### ISSUE TITLE
fix flakiness in test-coveyor 

### ISSUE THREAD

**Issue by maany** [rucio/rucio#5190]
Motivation
----------
A lot of integration tests fail due to the `test-cenveyor` failing in the pipeline, even when the files modified have nothing to do with the test.
```
_ test_multihop_intermediate_replica_lifecycle[file_config_mock1-caches_mock0-core_config_mock0] _
[gw0] linux -- Python 3.6.8 /usr/bin/python
test_conveyor.py:209: in test_multihop_intermediate_replica_lifecycle
    assert __get_source(request_id=request['id'], src_rse_id=jump_rse_id, **did).ranking == -1
E   assert 0 == -1
E     +0
E     --1
```
The goal is to fix the flakiness in the test.

**Comment by rcarpa** [comment]
Should be fixed by https://github.com/rucio/rucio/pull/5218

**Comment by maany** [comment]
Re-opening as we spotted the test failing again:
https://github.com/rucio/rucio/runs/5096512742?check_suite_focus=true

**Comment by maany** [comment]
Ahh, so I did not follow the extra commits from @rcarpa after re-opening the issue last week. Thanks for the patches @rcarpa. Since the new patches which were merged 6 days ago, there has been only 1 incident where the `test-conveyor` has failed in the integration tests:
https://github.com/rucio/rucio/pull/5239

I think the changes are independent of the conveyor? Can someone verify this? @faluchet @bari12 @rcarpa @joeldierkes  

If so, we should still consider this test flaky and I'll re-open this issue and send a PR to disable it. Thanks!

**Comment by rcarpa** [comment]
The issue is still persistent. But much less frequent. The issue can occur if the transfer of the second hop submitted here
https://github.com/rucio/rucio/blob/4ed5a2f90ffd5e3bd2b000aa6093910f7268d36e/lib/rucio/tests/test_conveyor.py#L184
does not finish executing before submitter is called a second time here;
https://github.com/rucio/rucio/blob/4ed5a2f90ffd5e3bd2b000aa6093910f7268d36e/lib/rucio/tests/test_conveyor.py#L203

I "assume" that a fix would be to wait for the actual execution to be "sure" that it is either done or failed. By adapting the function used here
https://github.com/rucio/rucio/blob/4ed5a2f90ffd5e3bd2b000aa6093910f7268d36e/lib/rucio/tests/test_conveyor.py#L1038
to be able to wait for any of the two states

**Comment by rcarpa** [comment]
But I still have a nasty feeling about it that it can be a real bug in the submitter.

**Comment by rcarpa** [comment]
This is fixed in 1.28
